### PR TITLE
[Telemetry] Enable telemetry tests for php, and V2 for python

### PR DESCRIPTION
**Pull request by paullegranddc**
## Description

* Split Telemetry v1 and v2 tests in two classes because fo different release dates/language support
* Enable telemetry V1 and V2 tests for Php
* Enable telemetry V2 tests for python
* Add tests for empty collections

## Workflow

1. ⚠️⚠️ Create your PR as draft (we're receiving lot of PR, it saves us lot of time) ⚠️⚠️
2. Follows the style guidelines of this project (See [how to easily lint the code](https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md))
3. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
4. Mark it as ready for review

> **_NOTE:_**  By default in PR only default scenario tests will be launched. Please refer to the [documentation](https://datadoghq.atlassian.net/wiki/spaces/APMINT/pages/2866381467/CI+Workflow+Github+Actions) to run all scenarios in your PR if needed.

Once your PR is reviewed, you can merge it ! :heart:

### CODE REVIEW

**Comment by cbeauchesne** [comment]
Can you want for #1465 being merged before merging this one? it'll activate those scenarios for all tracers

**Comment by paullegranddc** [comment]
I re-put this PR in draft because I just realised some tests were previously still skipped, and I might to adjust a few things before merging.

**Comment by cbeauchesne** [comment]
FYI, I skipped the test entirely, do not forget to remove the irrelevant : https://github.com/DataDog/system-tests/pull/1077/files#diff-6ab9dd4d058d45bd9344fb44a8b55b5c358f9e2648bf50bbdedcbfe2dd50220cR554

File: tests/test_telemetry.py [hunk 1]
```diff
@@ -496,13 +522,43 @@ def test_app_product_change(self):
...
-@released(python="1.7.0", dotnet="2.12.0", java="0.108.1", nodejs="3.2.0", ruby="1.4.0")
-@bug(context.uds_mode and context.library < "nodejs@3.7.0")
-@missing_feature(library="cpp")
-@missing_feature(library="php")
+@released(cpp="?", dotnet="2.35.0", golang="1.49.1", java="?", python="1.17.3", nodejs="?", php="0.90", ruby="1.11")
+class Test_TelemetryV2:
+    """Test telemetry v2 specific constraints"""
+
+    @missing_feature(library="golang", reason="Product started missing")
+    @missing_feature(library="dotnet", reason="Product started missing")
```
Comment (andrewlock) [inline tests/test_telemetry.py:530]: This feature should be working for .NET 🤔

File: tests/test_telemetry.py [hunk 2]
```diff
@@ -514,19 +570,21 @@ def test_app_started_product_disabled(self):
...
+                continue
+            payload = data["request"]["content"]["payload"]
 
-                for product, details in payload["products"].items():
-                    assert (
-                        details.get("enabled") is False
-                    ), f"Product information expected to indicate {product} is disabled, but found enabled"
+            assert (
+                "products" in payload
+            ), f"Product information was expected in app-started event, but was missing in {data['log_filename']}"
```
Comment (andrewlock) [inline tests/test_telemetry.py:579]: This is only expected in v2 telemetry - should we make that clearer?

**Review by andrewlock — APPROVED** [review]

File: tests/test_telemetry.py (hunk 1 above)
Comment (paullegranddc) [reply tests/test_telemetry.py:530]: Looks like this test is looking at `application.products`, when dotnet puts the data in `payload.products` . The ProductDisabled scenario does the opposite 🤷  which I missed.
I don't remember exactly why we have both... After checking we deserialize both value and merge them in the backend.

I really want to be done with this PR, so once this is merged, I'll open a second PR that looks at both locations for products for this test and the ProductDisabled scenario, and remove the annotation.
Is that fine?

Comment (andrewlock) [reply tests/test_telemetry.py:530]: >Looks like this test is looking at `application.products`,

[That's the "v1" location](https://github.com/DataDog/instrumentation-telemetry-api-docs/blob/main/GeneratedDocumentation/ApiDocs/v1/SchemaDocumentation/Schemas/application.md#application) 🙂 

[In "v2" it _only_ should be on `payload.products`](https://github.com/DataDog/instrumentation-telemetry-api-docs/blob/main/GeneratedDocumentation/ApiDocs/v2/SchemaDocumentation/Schemas/app_started.md) 

> I really want to be done with this PR, so once this is merged, I'll open a second PR that looks at both locations for products for this test and the ProductDisabled scenario, and remove the annotation. Is that fine?

Sure, whatever works for you 😄
